### PR TITLE
modify artifact copy api to support cosign

### PR DESCRIPTION
**Pull request by wy65701436**
Signed-off-by: Wang Yan <wangyan@vmware.com>

### CODE REVIEW

File: src/controller/artifact/controller.go [hunk 1]
```diff
@@ -140,23 +141,33 @@ type controller struct {
...
+	Accs []accessorymodel.AccessoryData
+}
+
+func (c *controller) Ensure(ctx context.Context, repository, digest string, option *ArtOption) (bool, int64, error) {
 	created, artifact, err := c.ensureArtifact(ctx, repository, digest)
 	if err != nil {
 		return false, 0, err
 	}
-	for _, tag := range tags {
+	for _, tag := range option.Tags {
```
Comment (chlins) [inline src/controller/artifact/controller.go:154]: do we need check option == nil ?

File: src/controller/artifact/controller.go [hunk 2]
```diff
@@ -140,23 +141,33 @@ type controller struct {
...
+		}
+	}
 	// fire event
 	e := &metadata.PushArtifactEventMetadata{
 		Ctx:      ctx,
 		Artifact: artifact,
 	}
-	if len(tags) > 0 {
-		e.Tag = tags[0]
+	if len(option.Tags) > 0 {
```
Comment (chlins) [inline src/controller/artifact/controller.go:169]: same as above

**Review by chlins — APPROVED** [review]
lgtm

**Review by heww — APPROVED** [review]

File: src/controller/artifact/controller.go (hunk 1 above)
Comment (wy65701436) [reply src/controller/artifact/controller.go:154]: updated

**Pull request merged**